recipes-kernel/kernel-tests/kernel-tests-files/kernel_dmesg_diff.py

```diff
@@ -304,6 +308,10 @@ def parse_args():
     device_desc = get_device_desc()
     previous_dmesg_record = get_previous_dmesg_record(db, kernel_version, os_version.major_minor, device_desc, logger)
 
+if args.basis_log_db_date:
+    previous_dmesg_record = get_dmesg_record_by_date(db, args.basis_log_db_date, logger)
```

Comment:
Rather than set previous_dmesg_record above (in the if/else cases of args.current_log_db_date) and then just overwrite it here, you should add an else case here to set previous_dmesg_record=get_dmesg_record_by_date(...). This structure will make this if/else mimic the one above, just for basis_log_db_date instead of current_log_db_date.

Reply:
I have cleaned this up now.